### PR TITLE
[SPARK-14294] [SQL] native support alter table ... rename to ...

### PR DESCRIPTION
**Pull request by bomeng**
## What changes were proposed in this pull request?

This PR addresses one of the tokens defined in SPARK-14129 (TOK_ALTERTABLE_RENAME):

Supports native execution of ALTER TABLE .. RENAME TO ... command

Syntax:

`ALTER TABLE table_name RENAME TO new_table_name;`
## How was this patch tested?

New tests were added to DDLSuite.scala and SQLQuerySuite.scala, as well as updated DDLCommandSuite.scala.

### CODE REVIEW

**Comment by hvanhovell** [comment]
ok to test

**Comment by hvanhovell** [comment]
@bomeng could you add the SQL tag to the PR title. This makes it easier for others to scan the PR queue.

Overall this looks pretty solid. Could you rebase/merge to the latest master (the ANTLR 3 parser has been removed today)?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 1]
```diff
@@ -158,5 +159,22 @@ class DDLSuite extends QueryTest with SharedSQLContext {
...
+  test("SPARK-14129: alter table - rename") {
+    import testImplicits._
+
+    withTempPath { dir =>
+      val path = dir.getCanonicalPath
+      val df =
+        sparkContext.parallelize(1 to 10).toDF("num")
+      df.write.format("parquet").save(path)
+      // create the temporary table
+      sql(s"CREATE TEMPORARY TABLE test1 USING parquet OPTIONS (path '$path')")
```
Comment (hvanhovell) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: Lets not create a parquet file. It is easer to use `org.apache.spark.sql.sources.DDLScanSource` (test class). This is easier to setup and isn't writing to disk. This would become somthing like:

    CREATE TEMPORARY TABLE test1 
    USING org.apache.spark.sql.sources.DDLScanSource
    OPTIONS (From '1', To '10', Table 'Test')

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 2]
```diff
@@ -158,5 +159,22 @@ class DDLSuite extends QueryTest with SharedSQLContext {
...
+      assert(sql("SHOW TABLES").collect().count(_ == Row("test1", true)) == 1)
```
Comment (hvanhovell) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: This is quite hard to follow. Please use `checkAnswer(sql(...), ...)` here and everywhere else (see the other tests in this file for examples).

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 3]
```diff
@@ -1811,4 +1811,18 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
...
     }
   }
+
+  test("SPARK-14129: alter table - rename") {
+    // make sure table test1 / test2 does not exist
+    sql("DROP TABLE test1")
+    sql("DROP TABLE test2")
+    // create the table
+    sql("CREATE TABLE test1 (i INT)")
+    assert(sql("SHOW TABLES").collect().count(_ == Row("test1", false)) == 1)
```
Comment (hvanhovell) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Same comment as in `DDLSuite.scala`.

**Comment by andrewor14** [comment]
@bomeng I happen to be working on the broader issue SPARK-14129. Would you mind if I take over this patch and build mine on top of your changes here?

**Comment by bomeng** [comment]
Not a problem. @andrewor14 Hope this solution help. Thanks.

**Comment by bomeng** [comment]
@hvanhovell I've updated the codes based on your comments. I will let @andrewor14 take over my codes for his task. Thanks.

**Comment by andrewor14** [comment]
Thanks @bomeng, I will post the PR shortly.
